docs/index.html

```diff
@@ -1556,35 +1558,87 @@ <h6 id="name-semantics-9">
+            <h5 id="name-array-access-2">
+<a href="#section-2.6.2.4" class="section-number selfRef">2.6.2.4. </a><a href="#name-array-access-2" class="section-name selfRef">Array Access</a>
             </h5>
 <section id="section-2.6.2.4.1">
               <h6 id="name-syntax-10">
 <a href="#name-syntax-10" class="section-name selfRef">Syntax</a>
               </h6>
-<p id="section-2.6.2.4.1-1">An array index is a union element consisting of an integer (in base 10).<a href="#section-2.6.2.4.1-1" class="pilcrow">¶</a></p>
+<p id="section-2.6.2.4.1-1">Array access is a union element which selects zero or more elements of an array.
+                        It does not match a node which is not an array.
+                        Array access takes the form of an index, which selects at most one element, or a slice, which selects zero or more elements.<a href="#section-2.6.2.4.1-1" class="pilcrow">¶</a></p>
 <div id="section-2.6.2.4.1-2">
 <pre class="sourcecode lang-abnf">
-union-element =/ integer
+union-element =/ array-index / array-slice
+</pre><a href="#section-2.6.2.4.1-2" class="pilcrow">¶</a>
+</div>
+<p id="section-2.6.2.4.1-3">An array index is a union element consisting of an integer (in base 10).<a href="#section-2.6.2.4.1-3" class="pilcrow">¶</a></p>
+<div id="section-2.6.2.4.1-4">
+<pre class="sourcecode lang-abnf">
+array-index = integer
+
 integer = [%x2D] (%x30 / (%x31-39 *%x30-39))
                             ; optional - followed by 0 or
                             ; sequence of digits with no leading zero
-</pre><a href="#section-2.6.2.4.1-2" class="pilcrow">¶</a>
+</pre><a href="#section-2.6.2.4.1-4" class="pilcrow">¶</a>
 </div>
-<p id="section-2.6.2.4.1-3">Note: the syntax does not allow integers with leading zeros such as <code>01</code> and <code>-01</code>.<a href="#section-2.6.2.4.1-3" class="pilcrow">¶</a></p>
+<p id="section-2.6.2.4.1-5">Note: the syntax does not allow integers with leading zeros such as <code>01</code> and <code>-01</code>.<a href="#section-2.6.2.4.1-5" class="pilcrow">¶</a></p>
+<p id="section-2.6.2.4.1-6">An array slice is a union element consisting of two or three integers (in base 10 and which may be omitted) separated by colons.<a href="#section-2.6.2.4.1-6" class="pilcrow">¶</a></p>
+<div id="section-2.6.2.4.1-7">
+<pre class="sourcecode lang-abnf">
+array-slice = [ integer ] ws %x3A ws [ integer ]
+                   [ ws %x3A ws non-zero-integer ]
+                            ; start:end or start:end:step
+
+non-zero-integer = [%x2D] %x31-39 *%x30-39
+                            ; optional - followed by
+                            ; sequence of digits with no leading zero
+</pre><a href="#section-2.6.2.4.1-7" class="pilcrow">¶</a>
+</div>
+<p id="section-2.6.2.4.1-8">Note: if the third integer (the <code>step</code> value) is present, it MUST be non-zero.<a href="#section-2.6.2.4.1-8" class="pilcrow">¶</a></p>
```

Comment:
> it MUST be non-zero

I'm not sure if we should make zero step illegal; maybe it's better to define that array-slice with zero step just matches nothing? I don't feel that this situation is worth complicationg syntax / throwing a syntax error; it's just like selecting an index out of range - so why we would want to especially handle this situation?

Another argument is query templates: user may use something like `sprintf("$.a[%d:%d:%d]", x, y, z)` and it may be quite surprising to get syntax errors on zero `z` and not getting it on any other value.